nodelet/pkg/phases/misc/misc_phase.go

```diff
@@ -1,20 +1,128 @@
 package misc
 
 import (
-	"path"
+	"context"
+	"fmt"
 
-	bashscript "github.com/platform9/nodelet/nodelet/pkg/phases/bash_script_based_phases"
+	"github.com/pkg/errors"
+	"github.com/platform9/nodelet/nodelet/pkg/utils/config"
 	"github.com/platform9/nodelet/nodelet/pkg/utils/constants"
+	"github.com/platform9/nodelet/nodelet/pkg/utils/kubeutils"
+	"github.com/platform9/nodelet/nodelet/pkg/utils/netutils"
+	"github.com/platform9/nodelet/nodelet/pkg/utils/phaseutils"
 	sunpikev1alpha1 "github.com/platform9/pf9-qbert/sunpike/apiserver/pkg/apis/sunpike/v1alpha1"
+
+	"go.uber.org/zap"
 )
 
-func NewMiscPhase(baseDir string) *bashscript.Phase {
-	miscPhase := &bashscript.Phase{
-		Filename: path.Join(baseDir, "misc_scripts.sh"),
+type MiscPhase struct {
+	HostPhase *sunpikev1alpha1.HostPhase
+	log       *zap.SugaredLogger
+	kubeUtils kubeutils.Utils
+	netUtils  netutils.NetInterface
+}
+
+func NewMiscPhase() *MiscPhase {
+	log := zap.S()
+	return &MiscPhase{
 		HostPhase: &sunpikev1alpha1.HostPhase{
 			Name:  "Miscellaneous scripts and checks",
 			Order: int32(constants.MiscPhaseOrder),
 		},
+		log:       log,
+		kubeUtils: nil,
+		netUtils:  netutils.New(),
+	}
+}
+
+func (m *MiscPhase) GetHostPhase() sunpikev1alpha1.HostPhase {
+	return *m.HostPhase
+}
+
+func (m *MiscPhase) GetPhaseName() string {
+	return m.HostPhase.Name
+}
+
+func (m *MiscPhase) GetOrder() int {
+	return int(m.HostPhase.Order)
+}
+
+func (m *MiscPhase) Status(ctx context.Context, cfg config.Config) error {
+
+	m.log.Infof("Running Status of phase: %s", m.HostPhase.Name)
+
+	if cfg.ClusterRole == constants.RoleMaster {
+		return nil
+	}
```

Comment:
I think we can get rid of this condition. The status checks being performed in this function are relevant on the master nodes as well. Lets try it without this if condition, if it fails then lets put it back in to unblock for now.